newsfragments/4219.added.md

```diff
@@ -0,0 +1,3 @@
+- Added `as_super` methods to `PyRef` and `PyRefMut` for accesing the base class by reference
+- Updated user guide to recommend `as_super` for referencing the base class instead of `as_ref`
```

Comment:
No need to include docs changes or internal changes here, but as this PR is otherwise great let's just leave these points here and I'll tidy up during release.

Reply:
Good to know, I'll keep that in mind next time. Thanks for the merge!